### ISSUE TITLE
RFC6186 : autodécouverte des services de courriels

### ISSUE THREAD

**Issue by lelutin** [AlternC/AlternC#250]
Salut!

Récemment chez Koumbit on a intégré les entrées DNS suggérées par RFC6186. Leur but c'est d'aider certains clients de courriels à découvrir les services disponibles et le hostname associé.

Voilà l'exemple des entrées dans la zone koumbit.net:

```
; RFC6186 records for mail service discovery by our users' MUAs.
non-explicit TLS services have lower priority
_submission._tcp IN SRV 0 1 587 smtp.koumbit.net.
_imap._tcp IN SRV 10 1 143 imap.koumbit.net.
_imaps._tcp IN SRV 0 1 993 imap.koumbit.net.
_pop3._tcp IN SRV 10 1 110 pop.koumbit.net.
_pop3s._tcp IN SRV 0 1 995 pop.koumbit.net.
```

Ça serait bien d'ajouter ces entrées là par défaut sur tous les domaines créés par alternc.

**Comment by sistemascta** [comment]
Just a heads-up: RFC6186 is not yet supported on Thunderbird, there's an open ticket that has been there for 12 years (!) with no signs of improvement: https://bugzilla.mozilla.org/show_bug.cgi?id=342242 . [This](https://bugzilla.mozilla.org/show_bug.cgi?id=342242#c56) and [this](https://bugzilla.mozilla.org/show_bug.cgi?id=342242#c59) comments are spot-on on mozilla's excuses about delaying its implementation.

**Comment by lelutin** [comment]
@sistemascta yeah thunderbird went with some other mechanism :\

but I figured if there are other clients that support this, then it'd give one more tool to make things easier for users. also it's only a set of 5 dns records so it's not very complex..

although thinking about it some more now, these records need to be setup automatically on a zone only if its emails are hosted on the alternc instance. so maybe they should be associated somehow to the setup of the MX in the zone.

**Comment by vincib** [comment]
good idea, (although I fear nobody will ever use those :/ microsoft and thundebird have their own autodiscover/autoconfig subdomains with XML files served through HTTPS, that AlternC supports since long ;) ) 
will try to merge this properly during 3.5.0rc2 !

**Comment by fulax** [comment]
As a follow-up to the issue title (not OP), the autoconfig/autodiscover mechanism have issues with HTTPS, as the subdomains autoconfig.* does not have a certificate yet.

However, [Thunderbird](https://wiki.mozilla.org/Thunderbird:Autoconfiguration) also checks `https?://domain.example/.well-known/autoconfig/mail/config-v1.1.xml` URIs.

Maybe we could add an Alias directive to catch those URLs and apply the rewrite rules ?
